### PR TITLE
Update spaceflights tutorial to take advantage of modular pipeline features

### PR DESCRIPTION
**Pull request by datajoely**
## Description
Spaceflights tutorial needs updating to demonstrate latest feature of kedro 

## Checklist

- [x] Read the [contributing](https://github.com/quantumblacklabs/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/quantumblacklabs/kedro/blob/main/RELEASE.md) file
- [x] Added tests to cover my changes

### CODE REVIEW

**Comment by datajoely** [comment]
Dear reviewer - the most useful QA you can provide is to run through the tutorial and to make sure that the snippets work.

File: .gitignore
Comment (datajoely) [inline .gitignore:159]: This is generated from running `make devserver` which builds a version of the docs you can read locally - we should exclude it from accidentally being committed

File: docs/source/03_tutorial/04_create_pipelines.md
Comment (datajoely) [inline docs/source/03_tutorial/04_create_pipelines.md:235]:
```suggestion
  filepath: data/02_intermediate/preprocessed_companies.pq
```

File: docs/source/03_tutorial/04_create_pipelines.md
Comment (datajoely) [inline docs/source/03_tutorial/04_create_pipelines.md:239]:
```suggestion
  filepath: data/02_intermediate/preprocessed_shuttles.pq
```

File: docs/source/03_tutorial/04_create_pipelines.md
Comment (datajoely) [inline docs/source/03_tutorial/04_create_pipelines.md:319]:
```suggestion
  filepath: data/03_primary/model_input_table.pq
```

File: docs/source/03_tutorial/04_create_pipelines.md
Comment (MerelTheisenQB) [inline docs/source/03_tutorial/04_create_pipelines.md:483]: I know this isn't your change, but should't we just merge these two blocks so you can copy it all at once?

File: docs/source/03_tutorial/04_create_pipelines.md
Comment (MerelTheisenQB) [inline docs/source/03_tutorial/04_create_pipelines.md:556]: Again not your change, but the above step isn't required if the user just followed the tutorial and ran `kedro pipeline create data_science`, I'd suggest removing it as it's a bit confusing.

File: docs/source/03_tutorial/05_namespacing_pipelines.md
Comment (MerelTheisenQB) [inline docs/source/03_tutorial/05_namespacing_pipelines.md:7]: It makes sense to add this sentence in this section, but it's not really one of the points being covered as far as I can tell. Perhaps just move it the next paragraph?

File: docs/source/03_tutorial/05_namespacing_pipelines.md
Comment (MerelTheisenQB) [inline docs/source/03_tutorial/05_namespacing_pipelines.md:19]: Previously we assumed the project was called `kedro_tutorial` and now it's `spaceflights_tutorial`, copy pasting the code below will lead to import errors.

**Review by MerelTheisenQB — APPROVED** [review]
Nice one @datajoely 👏 ! I went through all the steps of the tutorial to make sure the code runs.

File: docs/source/03_tutorial/04_create_pipelines.md
Comment (datajoely) [inline docs/source/03_tutorial/04_create_pipelines.md:27]:
```suggestion
    ├── kedro_tutorial
```

File: docs/source/03_tutorial/05_namespacing_pipelines.md
Comment (datajoely) [inline docs/source/03_tutorial/05_namespacing_pipelines.md:19]:
```suggestion
* Update the code in `src/kedro_tutorial/pipelines/data_processing/pipeline.py` using the snippet below.
```

File: docs/source/03_tutorial/05_namespacing_pipelines.md
Comment (datajoely) [inline docs/source/03_tutorial/05_namespacing_pipelines.md:7]:
```suggestion
```
I'm just going to remove it as it's actually not important

File: docs/source/03_tutorial/04_create_pipelines.md
Comment (datajoely) [inline docs/source/03_tutorial/04_create_pipelines.md:223]:
```bash
kedro viz
```

File: docs/source/03_tutorial/04_create_pipelines.md
Comment (datajoely) [reply docs/source/03_tutorial/04_create_pipelines.md:483]: Great point!

File: docs/source/03_tutorial/04_create_pipelines.md
Comment (datajoely) [reply docs/source/03_tutorial/04_create_pipelines.md:298]: Done!

File: docs/source/03_tutorial/04_create_pipelines.md
Comment (datajoely) [reply docs/source/03_tutorial/04_create_pipelines.md:246]: Updated

File: docs/source/03_tutorial/04_create_pipelines.md
Comment (datajoely) [reply docs/source/03_tutorial/04_create_pipelines.md:559]: I'm not sure what this is referring to - perhaps GH being weird. Do you remember?

Comment (MerelTheisenQB) [reply docs/source/03_tutorial/04_create_pipelines.md:559]: It was the step about creating an `__init__.py` file "As with the data processing modular pipeline, you should ensure that the data science modular pipeline is a Python package. To do so, create the file src/kedro_tutorial/pipelines/data_science/__init__.py containing the following:"

Comment (datajoely) [reply docs/source/03_tutorial/04_create_pipelines.md:553]: Done!

File: docs/source/03_tutorial/04_create_pipelines.md
Comment (idanov) [inline docs/source/03_tutorial/04_create_pipelines.md:250]:
```suggestion
[Apache Parquet](https://github.com/apache/parquet-format) is our recommended format for working with processed, typed data. In general we recommend you get your data out of CSV as soon as possible, Parquet supports things like compression, partitioning and types out of the box. Whilst you do lose the ability to view the file as text in our opinion the benefits greatly outweigh the drawbacks.
```

File: docs/source/index.rst
Comment (datajoely) [reply docs/source/index.rst:77]: It will - but we'll have to just accept that for the time being

File: docs/source/03_tutorial/05_namespacing_pipelines.md
Comment (datajoely) [reply docs/source/03_tutorial/05_namespacing_pipelines.md:19]: So strong disagreement here :) Also experience from VS Code is that it struggles to import `pipeline` correctly it gets confused between the package and method, so I found it easier to be explicit.

File: docs/source/03_tutorial/05_namespacing_pipelines.md
Comment (datajoely) [reply docs/source/03_tutorial/05_namespacing_pipelines.md:64]: Yes especially since we do the same for `data_science`!

File: docs/source/03_tutorial/05_namespacing_pipelines.md
Comment (datajoely) [reply docs/source/03_tutorial/05_namespacing_pipelines.md:210]: I actually think this may confuse things - we'd have to be very careful with our wording

File: docs/source/03_tutorial/05_namespacing_pipelines.md
Comment (datajoely) [reply docs/source/03_tutorial/05_namespacing_pipelines.md:198]: I can remove, but I think it makes more sense when rendered
image

WDYT?

Comment (idanov) [reply docs/source/03_tutorial/05_namespacing_pipelines.md:198]: I still think it is quite unclear for someone new to this.

File: docs/source/03_tutorial/05_namespacing_pipelines.md
Comment (idanov) [inline docs/source/03_tutorial/05_namespacing_pipelines.md:201]:
```suggestion
| :code:`inputs`     | Declaring ``model_input_table`` as a boundary input will not prefix it with the namespace | Same as `ds_pipeline_1`                    |
```
Maybe this makes it a bit clearer?

File: docs/source/03_tutorial/05_namespacing_pipelines.md
Comment (idanov) [inline docs/source/03_tutorial/05_namespacing_pipelines.md:203]:
```suggestion
| :code:`outputs`    | No outputs are at the boundary of this pipeline so nothing to list here                                      | Same as `ds_pipeline_1`                    |
```

File: docs/source/03_tutorial/05_namespacing_pipelines.md
Comment (idanov) [inline docs/source/03_tutorial/05_namespacing_pipelines.md:194]:
```suggestion
The table below describes what the purpose is of the keyword arguments in detail:
```

**Review by idanov — APPROVED** [review]

File: docs/source/03_tutorial/05_namespacing_pipelines.md
Comment (datajoely) [reply docs/source/03_tutorial/05_namespacing_pipelines.md:19]: Do you want to make the final call here?

File: docs/source/03_tutorial/05_namespacing_pipelines.md
Comment (idanov) [reply docs/source/03_tutorial/05_namespacing_pipelines.md]: Sorry, this one is not gramatically correct after my addition... I think we should move `is` after `arguments`...

File: docs/source/03_tutorial/05_namespacing_pipelines.md
Comment (datajoely) [inline docs/source/03_tutorial/05_namespacing_pipelines.md:194]:
```suggestion
The table below describes what the purpose of each keyword arguments in detail:
```

File: docs/source/03_tutorial/05_namespacing_pipelines.md
Comment (datajoely) [reply docs/source/03_tutorial/05_namespacing_pipelines.md]: You're right I've updated is as follows:
> The table below describes the purpose of each keyword arguments in detail

File: docs/source/03_tutorial/05_namespacing_pipelines.md
Comment (yetudada) [inline docs/source/03_tutorial/05_namespacing_pipelines.md:8]:
```suggestion
Adding namespaces to [modular pipelines](https://kedro.readthedocs.io/en/stable/06_nodes_and_pipelines/03_modular_pipelines.html#modular-pipelines) unlocks some sophisticated functionality in Kedro.
```

File: docs/source/03_tutorial/05_namespacing_pipelines.md
Comment (yetudada) [inline docs/source/03_tutorial/05_namespacing_pipelines.md:19]:
```suggestion
* This introduces the lowercase `pipeline()` method imported from `kedro.pipeline.modular_pipeline`.
```

File: docs/source/03_tutorial/05_namespacing_pipelines.md
Comment (yetudada) [inline docs/source/03_tutorial/05_namespacing_pipelines.md:75]:
```suggestion
* When introducing a namespace you must tell Kedro which inputs/outputs live at the 'edges' of the namespace.
```

**Review by yetudada — COMMENTED** [review]
At a high-level I'm glad we finally have a tutorial taking users through modular pipelines but it has made me quite nervous because it introduces so many new things to users e.g. "namespacing". If I'm a junior data scientist I suspect I'll get lost. 

I also note that we do increase complexity for end users; one of the pieces of feedback we have about Kedro is "Why do I have to touch so many files?" and the use of `kedro pipeline create` pushes users to even more files because we expect them to edit seperate parameters files. 

So I think we can merge this but we should definitely see if we can test this with new users through recorded sessions and then based on that we should regroup about how to reduce complexity again.

**Comment by datajoely** [comment]
> At a high-level I'm glad we finally have a tutorial taking users through modular pipelines but it has made me quite nervous because it introduces so many new things to users e.g. "namespacing". If I'm a junior data scientist I suspect I'll get lost.
> 
> I also note that we do increase complexity for end users; one of the pieces of feedback we have about Kedro is "Why do I have to touch so many files?" and the use of `kedro pipeline create` pushes users to even more files because we expect them to edit seperate parameters files.
> 
> So I think we can merge this but we should definitely see if we can test this with new users through recorded sessions and then based on that we should regroup about how to reduce complexity again.

I think this is super valid - I guess my intention here is to nudge users to use this pattern from day 1 so they dont create their own folder structures that are harder to manage (and micro-package)

File: docs/source/03_tutorial/01_spaceflights_tutorial.md
Comment (datajoely) [reply docs/source/03_tutorial/01_spaceflights_tutorial.md:19]: Removed

File: docs/source/03_tutorial/02_tutorial_template.md
Comment (datajoely) [reply docs/source/03_tutorial/02_tutorial_template.md:66]: Yes it absolutely is - I've included this so you don't get that enormous warning the first time you run things

File: docs/source/03_tutorial/04_create_pipelines.md
Comment (datajoely) [reply docs/source/03_tutorial/04_create_pipelines.md:20]: Done

File: docs/source/03_tutorial/04_create_pipelines.md
Comment (datajoely) [reply docs/source/03_tutorial/04_create_pipelines.md:354]: Done

File: docs/source/03_tutorial/04_create_pipelines.md
Comment (datajoely) [reply docs/source/03_tutorial/04_create_pipelines.md:20]: I can't get it to work - searching suggests we need an additional plugin
image

File: docs/source/03_tutorial/07_visualise_pipeline.md
Comment (datajoely) [reply docs/source/03_tutorial/07_visualise_pipeline.md]: Removed

File: docs/source/03_tutorial/06_namespacing_pipelines.md
Comment (datajoely) [reply docs/source/03_tutorial/06_namespacing_pipelines.md:184]: Done

**Pull request merged**
